### PR TITLE
Tang Packing

### LINKED ISSUE #21: Move "tang packing" from old repo to here
And make it compliant.

### PR DESCRIPTION
**Pull request by mmghannam**
Closes #21

### CODE REVIEW

File: geco/mips/packing.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    .. [1] Tang, Y., Agrawal, S., & Faenza, Y. (2019). Reinforcement learning for integer
+    programming: Learning to cut. arXiv preprint arXiv:1906.04859.
+    """
+    return packing(n, m, *tang_params(n, m, seed), binary, name="Tang Packing")
+
+
+@py_random_state(-1)
+def tang_params(n, m, seed=0):
+    costs = [seed.randint(1, 10) for _ in range(n)]
+    constraint_coefficients = [[seed.randint(0, 5) for _ in range(n)] for _ in range(m)]
```
Comment (CharJon) [inline geco/mips/packing.py]: randint(0,5+1)? (see above)

File: geco/mips/packing.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    ----------
+    .. [1] Tang, Y., Agrawal, S., & Faenza, Y. (2019). Reinforcement learning for integer
+    programming: Learning to cut. arXiv preprint arXiv:1906.04859.
+    """
+    return packing(n, m, *tang_params(n, m, seed), binary, name="Tang Packing")
+
+
+@py_random_state(-1)
+def tang_params(n, m, seed=0):
+    costs = [seed.randint(1, 10) for _ in range(n)]
```
Comment (CharJon) [inline geco/mips/packing.py:28]: randint is [a,b) isn't it? So we need randint(1,10+1)

File: geco/mips/packing.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    constraint_coefficients = [[seed.randint(0, 5) for _ in range(n)] for _ in range(m)]
+    limits = [seed.randint(9 * n, 10 * n) for _ in range(m)]
```
Comment (CharJon) [inline geco/mips/packing.py]: randint(9 * n, 10 * n +1)? (see above)

File: geco/mips/packing.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            number of constraints
+        seed: int, random state or None
+            seed for randomization
+
+    References:
+    ----------
+    .. [1] Tang, Y., Agrawal, S., & Faenza, Y. (2019). Reinforcement learning for integer
+    programming: Learning to cut. arXiv preprint arXiv:1906.04859.
+    """
+    return packing(n, m, *tang_params(n, m, seed), binary, name="Tang Packing")
```
Comment (CharJon) [inline geco/mips/packing.py]: Binary packing uses different params!

**Review by CharJon — CHANGES_REQUESTED** [review]

File: geco/mips/packing.py (hunk 2 above)
Comment (mmghannam) [reply geco/mips/packing.py:28]: it's [a,b] in the [docs](https://docs.python.org/3/library/random.html#random.randint)

Comment (CharJon) [reply geco/mips/packing.py:28]: Oh, mh. It's [a,b) for numpy. Is this a problem?

**Pull request merged**